torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -667,16 +701,33 @@ std::string TensorPipeAgent::getDefaultIPAddress() {
   return defaultIP;
 }
 
+void TensorPipeAgent::increaseCallCount(int32_t& count) {
+  std::unique_lock<std::mutex> lock(callCountMutex_);
+  ++count;
+  callCountCV_.notify_all();
```

Comment:
Do we need to unlock before notify all?

Reply:
There is no _need_ to unlock before notifying, but you're right that there's also no need to _hold_ the lock while notifying. In fact, in some cases it's more efficient to release the lock before notifying, as otherwise the thread that gets woken up would just immediately block again on acquiring the lock. However, pthreads optimizes for that and there's no performance penalty. The docs for `notify_one` has an entire paragraph dedicated to this: https://en.cppreference.com/w/cpp/thread/condition_variable/notify_one

In the end, I'm going to release the lock, it's cleaner.